polkadot/xcm/xcm-builder/src/barriers.rs

```diff
@@ -504,3 +506,36 @@ impl ShouldExecute for DenyReserveTransferToRelayChain {
 		Ok(())
 	}
 }
+
+// See https://github.com/paritytech/polkadot-sdk/pull/6838
+pub struct DenyFirstExportMessageFrom<FromOrigin, ToGlobalConsensus>(
+	PhantomData<(FromOrigin, ToGlobalConsensus)>,
+);
+
+impl<FromOrigin, ToGlobalConsensus> ShouldNotExecute
+	for DenyFirstExportMessageFrom<FromOrigin, ToGlobalConsensus>
+where
+	FromOrigin: Contains<Location>,
+	ToGlobalConsensus: Contains<NetworkId>,
+{
+	fn should_not_execute<RuntimeCall>(
+		origin: &Location,
+		message: &mut [Instruction<RuntimeCall>],
+		_max_weight: Weight,
+		_properties: &mut Properties,
+	) -> Result<(), ProcessMessageError> {
+		message.matcher().match_next_inst_while(
```

Comment:
I think that `match_next_inst_while` with `ControlFlow::Continue` checks all the instructions, so the name `DenyFirstExportMessageFrom` is not correct, so let's better use `DenyExportMessage`.

I am just thinking, that my initial advice `DenyFirst` was wrong, if there are multiple `ExportMessage` instructions, we should check them all for sure.

Reply:
https://github.com/paritytech/polkadot-sdk/pull/7169/commits/bf669bb24eb069f3ae5cfe9b7e9f75ac9d7850e3